### PR TITLE
added more robust generated file location handling

### PR DESCRIPTION
**Pull request by joedevivo**
Addresses issue #32

### CODE REVIEW

File: src/cuttlefish_escript.erl [hunk 1]
```diff
@@ -113,7 +113,15 @@ main(Args) ->
...
             DP;
         true ->
-             proplists:get_value(dest_dir, ParsedArgs)
+            DP = proplists:get_value(dest_dir, ParsedArgs),
+            case filelib:ensure_dir(filename:join(DP, "weaksauce.dummy")) of
+                %% filelib:ensure_dir/1 requires a dummy filename in the argument,
+                %% I think that is weaksauce, hence "weaksauce.dummy" 
+                ok -> DP;
+                {error, E} ->
+                    lager:error("Error (~p) creating ~s", [E, DP]),
```
Comment (jonmeredith) [inline src/cuttlefish_escript.erl]: After talking about logging with CSEs and candidate today I'm feeling pedantic about log messages

```
2013-10-02 13:13:45.250 [error] <Pid> Error (~p) creating ~s line 122
```

yodawg, I put some errors in your errors.

```
Unable to create directory ~s - ~p.  Please check permissions.
```

Comment (joedevivo) [inline src/cuttlefish_escript.erl]: done

File: src/cuttlefish_escript.erl [hunk 2]
```diff
@@ -166,6 +175,11 @@ filename_maker(Filename, Date, Extension) ->
             Extension
         ]).
 
+quitter(RC) ->
+    %% Let's give lager a chance to display some things.
+    timer:sleep(500),
+    halt(RC).
```
Comment (jonmeredith) [inline src/cuttlefish_escript.erl]: Why not use init:stop() ?  That would give lager a chance to shutdown nicely without the 0.5s pause.

**Comment by jaredmorrow** [comment]
From a generated file directory standpoint I'm +1, it worked with https://github.com/basho/node_package/pull/100

I would prefer if we could remove the sleep though.

**Pull request merged**